File: MTGLibs/src/main/java/com/common/control/base/BaseDialog.java
package com.common.control.base;

import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import androidx.fragment.app.DialogFragment;


public abstract class BaseDialog<T extends ViewDataBinding> extends DialogFragment {
    protected T binding;

    protected OnActionCallback callback;

    public void setCallback(OnActionCallback callback) {
        this.callback = callback;
    }

    public void showDialog(AppCompatActivity activity) {
        show(activity.getSupportFragmentManager(), null);
    }

    public void showDialog(AppCompatActivity activity, OnActionCallback callback) {
        setCallback(callback);
        show(activity.getSupportFragmentManager(), null);
    }

    @Nullable
    @Override
    public final View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, getLayoutId(), container, false);
        if (getDialog() != null && getDialog().getWindow() != null) {

            getDialog().getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
            getDialog().getWindow().requestFeature(Window.FEATURE_NO_TITLE);

        }
        prepareDialog();
        return binding.getRoot();
    }

    protected abstract void prepareDialog();

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initView();
        addEvent();
    }

    protected abstract int getLayoutId();

    protected abstract void initView();

    protected abstract void addEvent();


}

File: app/src/main/java/com/mtg/tool/findmyphone/consent_dialog/base/BaseDialog.kt
package com.mtg.tool.findmyphone.consent_dialog.base

import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.fragment.app.DialogFragment
import com.common.control.base.OnActionCallback
import com.mtg.tool.findmyphone.base.BaseActivity

abstract class BaseDialog<T : ViewDataBinding?> : DialogFragment() {
    @JvmField
    protected var binding: T? = null
    @JvmField
    protected var callback: OnActionCallback? = null
    var dismissListener: OnActionCallback? = null
    fun setCallback(callback: OnActionCallback?) {
        this.callback = callback
    }

    fun showDialog(activity: BaseActivity<*>) {
        show(activity.supportFragmentManager, null)
    }

    fun showDialog(activity: AppCompatActivity) {
        show(activity.supportFragmentManager, null)
    }

    fun showDialog(activity: AppCompatActivity, callback: OnActionCallback?) {
        setCallback(callback)
        show(activity.supportFragmentManager, null)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = DataBindingUtil.inflate(inflater, layoutId, container, false)
        return binding!!.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        if (dialog != null && dialog!!.window != null) {
            dialog!!.window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
            dialog!!.window!!.setLayout(
                WindowManager.LayoutParams.MATCH_PARENT,
                WindowManager.LayoutParams.WRAP_CONTENT
            )
        }
        initView()
        addEvent()
    }

    override fun onDestroy() {
        super.onDestroy()
        if (dismissListener != null) {
            dismissListener!!.callback(null, null)
        }
    }

    protected abstract val layoutId: Int
    protected abstract fun initView()
    protected abstract fun addEvent()
}